### ISSUE TITLE
No Networking in Qubes R4 AppVMs After Upgrade From R3.2

### ISSUE THREAD

**Issue by manasb** [QubesOS/qubes-issues#4075]
### Qubes OS version:
`R4`

### Affected component(s): 

AppVMs imported from Qubes R3.2 - This AppVM and the TemplateVM it is based on were imported from a R3.2 backup. `fedora-28` in this case

---

### Steps to reproduce the behavior:

- Install Qubes R3.2
- Upgrade included `fedora` template to `fedora-28`
- Networking functions normally in Qubes R3.2 - all AppVMs have expected networking access
- Backup AppVM
- Install Qubes R4
- Restore TemplateVMs and AppVMs from R3.2
- Change repository for TemplateVMs to qubes-4 (I imported this from the `fedora-26` template in Qubes R4 from the `/etc/yum.repos.d/` directory as well as the PGP key in the `/etc/pki/rpm-gpg` directory)
- Update all software in TemplateVM (my `fedora-28` TemplateVM is able to fetch updates through `dnf`)
- Ensure AppVM is set to use the `sys-firewall` NetVM (confirm both `sys-firewall` and `sys-net` can successfully make DNS queries and ping)
- Start AppVM
- `ifconfig` in AppVM only shows the loopback interface
- AppVM can not access the internet
- Qubes Manager however shows an IP, Netmask and Gateway for the AppVM

### Expected behavior:

- Functional networking in AppVM imported from a Qubes R3.2 backup

### Actual behavior:

- No network available in AppVMs imported from Qubes R3.2 backup

### General notes:

Found some networking-related scripts in the `/etc/xen` directory on the `fedora-26` TemplateVM that comes with R4 and copied them over to `fedora-28`, this did not help however.

---

**Comment by manasb** [comment]
I had used a `fedora-28` StandaloneVM which was upgraded, backed up & restored in a similar way to write that last post. Now, after a reboot no programs from that StandaloneVM are working, I am able to `sudo xl console <vm>` into the system but no programs will run and it too is now unable to access the network.

**Comment by marmarek** [comment]
Make sure you have `qubes-vm-recommended` package installed there (if you haven't it there originally, install during upgrade). Check [this](https://www.qubes-os.org/doc/templates/fedora-minimal/) for details about individual packages.

**Comment by manasb** [comment]
@marmarek 

Excellent! That fixed network connectivity for my AppVMs.
I'm thinking about how I could apply this to my StandaloneVM where the network doesn't work.. Any ideas?

**Comment by marmarek** [comment]
You need to setup network manually (add IP address, default route, DNS) according to values in VM settings (DNS address is 10.139.1.1).

**Comment by manasb** [comment]
Networking is functioning in my StandaloneVM again, thank you. I did not try your instructions, I was able to resolve it by downloading the packages through the TemplateVM, shifting them over, installing and restarting the VM.

I am not able to run any programs, however.
I have tried to do this through the Qubes Manager as well as with `qvm-run`. The command hangs and does not complete, a command such as `qvm-run <vm> konsole` does not start up the `konsole` terminal client.
